spec/controllers/client_api/users/users_controller_spec.rb

```diff
@@ -0,0 +1,223 @@
+require 'rails_helper'
+
+describe ClientApi::Users::UsersController, type: :controller do
+  login_user
+
+  before do
+    @user = User.first
+  end
+
+  describe 'GET current_user_info' do
+    it 'responds successfully' do
+      get :current_user_info, format: :json
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe 'POST change_password' do
+    it 'responds successfully' do
+      post :change_password,
+           params: { user: { password: 'secretPassword'} },
+           format: :json
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'changes password' do
+      expect(@user.valid_password?('secretPassword')).to eq(false)
+      post :change_password,
+           params: { user: { password: 'secretPassword'} },
+           format: :json
+
+      expect(@user.reload.valid_password?('secretPassword')).to eq(true)
+    end
+
+    it 'does not change short password' do
+      expect(@user.valid_password?('pass')).to eq(false)
+      post :change_password,
+           params: { user: { password: 'pass'} },
+           format: :json
+
+      expect(@user.reload.valid_password?('pass')).to eq(false)
+    end
+  end
+
+  describe 'POST change_timezone' do
+    it 'responds successfully' do
+      user = User.first
+      expect(user.time_zone).to eq('UTC')
+      post :change_timezone, params: { timezone: 'Pacific/Fiji'}, format: :json
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'changes timezone' do
+      user = User.first
+      expect(user.time_zone).to eq('UTC')
+      post :change_timezone, params: { timezone: 'Pacific/Fiji'}, format: :json
+      expect(user.reload.time_zone).to eq('Pacific/Fiji')
+    end
+  end
+
+  describe 'POST change_initials' do
+    it 'responds successfully' do
+      post :change_initials, params: { initials: 'TD'}, format: :json
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'responds successfully' do
+      user = User.first
+      expect(user.initials).not_to eq('TD')
+      post :change_initials, params: { initials: 'TD'}, format: :json
+      expect(user.reload.initials).to eq('TD')
+    end
+  end
+
+  describe 'POST change_system_notification_email' do
+    it 'responds successfully' do
+      post :change_system_notification_email,
+           params: { status: false },
+           format: :json
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'changes notification from false => true' do
+      user = User.first
+      user.system_message_notification_email = false
+      user.save
```

Comment:
do you need persistence here? It will slow down the tests

Reply:
We could fake|mock .save method but I think in this particular test this is not the best way. Saving would be an issue if this would trigger a lot of saves (if you are creating protocol and you have to create -> user -> team etc. in this case it would make sense to mock .save method). 

But because it is just one save in database per test this will not be an issue on performance. Also mocking can sometimes be pain because it could lead to false positive tests.

Reply:
Check if the test passes if you skip .save